src/v2/controllers/FormController.js

```diff
@@ -156,7 +161,22 @@ export default Controller.extend({
         .then(this.handleIdxSuccess.bind(this))
         .catch(error => {
           this.showFormErrors(this.formView.model, error);
+        })
+        .then(() => {
```

Comment:
This is going to create problems for us, as it appears we executing this handler after the resolved promise from the [`handleIdxResponseSuccess`](https://github.com/okta/okta-signin-widget/blob/55a5d9cad217477abe56de8ab5dd828546e54888/src/v2/BaseLoginRouter.js#L85) method.

We already check for the `actionPath` above (see line 141) - why can't this logic sit there?

Reply:
I assumed that first we need to cancel current login flow, and then after `/cancel` response we can  start a new one.

Reply:
But this logic can be moved to line 141

Reply:
Moved this logic instead of `this.handleIdxSuccess`  as suggested by @haishengwu-okta 
https://github.com/okta/okta-signin-widget/pull/2090#discussion_r668471099